cmd/jujud/updateseries/updateseries.go

```diff
@@ -103,8 +97,35 @@ func (c *UpdateSeriesCommand) SetFlags(f *gnuflag.FlagSet) {
 	f.BoolVar(&c.startAgents, "start-agents", false, "start agents if successful with update")
 }
 
+// GetJujuVersion gets the version of the agent from agent's config file
+func GetJujuVersion(machineAgent string, dataDir string) (version.Number, error) {
+	agentConf := agent.NewAgentConf(dataDir)
+	if err := agentConf.ReadConfig(machineAgent); err != nil {
+		return version.Number{}, err
+	}
+	config := agentConf.CurrentConfig()
+	if config == nil {
+		return version.Number{}, errors.Errorf("%s agent conf is not found", machineAgent)
+	}
+	return config.UpgradedToVersion(), nil
+}
+
 func (c *UpdateSeriesCommand) Run(ctx *cmd.Context) error {
-	if err := c.findAgents(ctx); err != nil {
+
+	var (
+		err                 error
+		errAgentNames       []string
+		errUnitNames        []string
+		errMachineName      string
+		errSysdServiceNames []string
+		errSymServiceNames  []string
+	)
+	c.manager = service.NewSystemdServiceManager(func() (bool, error) { return true, nil })
+
+	if c.machineAgent, c.unitAgents, errAgentNames, err = c.manager.FindAgents(c.dataDir); err != nil {
+		for _, name := range errAgentNames {
+			ctx.Warningf("%s is not of type Machine nor Unit, ignoring", name)
```

Comment:
move this loop to print the info out of the if err != nil, errAgentNames only returned if err == nil